keps/sig-instrumentation/1668-trace-popagating/README.md

```diff
@@ -0,0 +1,501 @@
+
+This mutating admission webhook extracts  a `span context` from incoming request, and then stores it into object annotation`trace.kubernetes.io/context` with base64 encoded version of [this wire format](https://github.com/census-instrumentation/opencensus-specs/blob/master/encodings/BinaryEncoding.md#trace-context). The webhook can be configured to inject context into only target object types.
+
+below is a key/value pair example in object annotation :
+
+| key               | value(encoded)                       | origin value(decoded)                                   | description                                                  |
+| ---------------------------- | ------------------------------------------------------- | ------------------------------------------------------------ | ------------------------------------------------------------ |
+| trace.kubernetes.io/context | 0/3kO3imLJzu3N54RTuOEUDbc2C0poIMAA== | 00-ca057eae1a26b66314fe3e361eedc5ca-3696483da6bfdcea-00 |it consists of `<version>-<traceid>-<spanid>-<flag>`. <br />A corresponding http header field is like `traceparent: 00-ca057eae1a26b66314fe3e361eedc5ca-3696483da6bfdcea-00` which is a w3c specific [trace-context](https://w3c.github.io/trace-context/#traceparent-header ). <br />|
+
+### Behaviors with and without Mutating webhook
+Since the mutating webhook is optional for users, we will explain the different behaviors between with and without mutating webhook.
+
+#### with Mutating webhook
+
+**kubectl request**:
+
+- APIServer uses otel to start a new Span
+- APIServer uses otel to propagate `span context` to the other end(webhook)
+- Webhook persists `span context` to object
+
+**controllers request:**
+
+- Controller uses otel to start a related Span, which connected to the `span context` in object
+- Controller uses otel to propagate `span context`  to the other end(APIServer)
+- APIServer uses otel  to start a related Span, which connected to the `span context` from the  incoming request
+- APIServer uses otel to propagate `span context`  to the other end(webhook)
+- Webhook persists `span context` to object
+
+#### without Mutating webhook
+
+**kubectl request:**
+
+- APIServer uses otel to start a new Span
+- APIServer uses otel to propagate `span context` to the other end
+- ~~Webhook persists `span context` to object~~
+
+**controllers request:**
+
+- Controller start uses otel to start a new Span
+- Controller uses otel to propagate `span context`  to the other end(APIServer)
+- APIServer uses otel  to start related Span, which connected to the `span context` from the  incoming request
+- APIServer uses otel to propagate `span context` to the other end
+- ~~Webhook persists `span context`  to object~~
+
+In short, the webhook decides whether to add `span context` to the object.
+
+### Test Plan
+
+All added code will be covered by unit tests.
```

Comment:
We should also have integration tests for the mutating admission controller

Reply:
and e2e tests probably

Reply:
Got it.